client/errors.go

```diff
@@ -8,6 +8,9 @@ import (
 // ErrConnectionFailed is an error raised when the connection between the client and the server failed.
 var ErrConnectionFailed = errors.New("Cannot connect to the Docker daemon. Is the docker daemon running on this host?")
 
+// ErrPermissionDenied is an error raised when a unix socket can not be opened.
+var ErrPermissionDenied = errors.New("Got permission denied while trying to connect to the Docker daemon.")
```

Comment:
How about Got permission denied while trying to connect to the Docker daemon's socket.